core/schema/service.go

```diff
@@ -18,9 +18,21 @@ var _ SchemaResolvers = &serviceSchema{}
 
 func (s *serviceSchema) Install() {
 	dagql.Fields[*core.Container]{
-		dagql.Func("asService", s.containerAsService).
+		dagql.Func("asService", s.containerAsServiceLegacy).
+			View(BeforeVersion("v0.13.8")).
```

Comment:
We didn't release a v0.13.8, next version is v0.14.1, but maybe this is supposed to be scheduled for v0.15.0?

Reply:
oh right. thanks for catching this.

Reply:
I added with version 0.14.0 for now. If I change this to version v0.15.0, it didn't work for me.

Reply:
> If I change this to version v0.15.0, it didn't work for me.

If you look at other tests in `core/integration/legacy_test.go` you can see that there's a version override for GraphQL queries (`testutil.QueryOptions{Version: "xxx"}`). You can use this to simulate a future version.

So `BeforeVersion` should target `v0.15.0`, and as for `AfterVersion`, it's useful if you want to merge before that release so you don't have to keep rebasing and potentially having to solve merge conflicts over and over.

But if you intend to hold on to the merge until the `v0.15.0` release, then just use `View(AllVersion)` for the new default, which will take effect for any version that doesn't match the `BeforeVersion`.

 See some current examples of this in the schema (like `schema/container.go`).

Reply:
specifying the engine version in `dagger.json` worked for me this time. I am using `v0.15.0` for both `AfterVersion` and `BeforeVersion` as we do intend to merge this before v0.15.0

@jedevc has targetted this issue for v0.14.1 as well, i am wondering if we need to handle that as well.

Reply:
Hi @helderco 

I have a question:

- so lets say current release is v0.14.0
- now I changed the AsService api so that for `BeforeVersion("v0.15.0")` it should serve old api, and for `AfterVersion("v0.15.0")` it serves the new version.
- now when we run the tests, it runs with the current version of dagger, which would be `v0.14.1-.....`
- that would mean that the api visible to tests would be the old one. so we don't need any changes in existing tests yet.
- now lets say `v0.15.0` is released. this would mean tests would suddently start seeing `v0.15.0` version of api, and the tests would fail as the new api is no longer backward compatible. 
- now we would need to make changes to the existing tests?

does ^^ makes sense? It almost feels like I am missing something obvious here.

Reply:
Oop, responded on discord: https://discord.com/channels/707636530424053791/1003718839739105300/1312018293812494357